drivers/frequency/ad9523/ad9523.c

```diff
@@ -235,7 +235,7 @@ int32_t ad9523_calibrate(struct ad9523_dev *dev)
 			AD9523_READBACK_1,
 			&reg_data);
 	if ((reg_data & 0x1) != 0x0) {
-		printf("AD9523: VCO calibration failed (%x)!\n", reg_data);
+		printf("AD9523: VCO calibration failed (0x%04lx)!\n", reg_data);
```

Comment:
just a suggestion,

instead of explicitly putting the 0x in front, you could add the # specifier after the % like:
%#04lx
which will add the 0x to the output for you.

Reply:
Thank you for the suggestion, this is very useful. The "0x" prefix is added, but it seems that this way, the padding does not work as expected.